### PR TITLE
GFORMS-1076 - Add a new expression to allow for the count of Add to L…

### PR DESCRIPTION
**Pull request by josef-vlach**
…ist items to be displayed

### CODE REVIEW

File: app/uk/gov/hmrc/gform/eval/EvaluationResults.scala [hunk 1]
```diff
@@ -112,7 +111,12 @@ case class EvaluationResults(
       case ctx @ FormCtx(formComponentId: FormComponentId) => get(ctx, recData, fromVariadicValue)
       case Sum(FormCtx(formComponentId)) =>
         calculateSum(formComponentId, recData, unsupportedOperation("Number")(expr))
-      case Sum(_)                                     => unsupportedOperation("Number")(expr)
+      case Sum(_) => unsupportedOperation("Number")(expr)
+      case Count(formComponentId) =>
+        val xs: Iterable[(ModelComponentId, VariadicValue)] =
+          recData.variadicFormData.forBaseComponentId(formComponentId.baseComponentId)
+        val zeros: Int = xs.map(_._2).count(_.contains(0.toString))
+        NumberResult(zeros + 1)
```
Comment (SandipKhedekar) [inline app/uk/gov/hmrc/gform/eval/EvaluationResults.scala:119]: Can we please look into code formating for better reading?

Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/eval/EvaluationResults.scala:119]: This is `scalafmt` formatted. What exactly is your concern?

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala [hunk 2]
```diff
@@ -45,6 +46,7 @@ sealed trait Expr extends Product with Serializable {
     case Else(field1: Expr, field2: Expr)           => field1.sums ++ field2.sums
     case FormCtx(formComponentId: FormComponentId)  => Nil
     case sum @ Sum(field1: Expr)                    => sum :: Nil
+    case Count(field1: FormComponentId)             => Nil
```
Comment (SandipKhedekar) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala:49]: **field1** never used.

File: app/uk/gov/hmrc/gform/eval/EvaluationResults.scala (hunk 1 above)
Comment (SandipKhedekar) [reply app/uk/gov/hmrc/gform/eval/EvaluationResults.scala:119]: Screen Shot 2021-02-02 at 09 53 41

wouldn't be easy to read at first instance.

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala (hunk 2 above)
Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala:49]: Yes, but it is consistent with rest of the definition. And someone can view it as a documentation for the reader.

If you are somehow concerned about doing that, let's take that to another discussion and let's do the change in one fell swoop across whole project

Comment (SandipKhedekar) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/Expr.scala:49]: I agree there are many places in the codebase so let it be inline.

File: app/uk/gov/hmrc/gform/eval/EvaluationResults.scala (hunk 1 above)
Comment (josef-vlach) [reply app/uk/gov/hmrc/gform/eval/EvaluationResults.scala]: Fixed

**Review by SandipKhedekar — APPROVED** [review]

**Pull request merged**